tests/libvmi/lifecycle.go

```diff
@@ -0,0 +1,100 @@
+package libvmi
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	vmiSet := VMISet{"vmi": vmi}
+	vmiSet.Setup()
+	return vmiSet["vmi"]
+}
+
+func CleanupVMI(vmi *v1.VirtualMachineInstance) {
+	vmiSet := VMISet{"vmi": vmi}
+	vmiSet.Cleanup()
+}
+
+type VMISet map[string]*v1.VirtualMachineInstance
```

Comment:
Can we put the namespace as part of the key with a struct or the like ? sometimes `NamespaceTestAlternative` is used.

Reply:
You can set the namespace name as part of the VMI spec if it will be needed.